### PR TITLE
chore: adapt upgrade e2e workflow for custom images

### PR DESCRIPTION
**Pull request by damiannolan**
## Description

- Removing cron expression in `e2e-upgrade.yml` 
- Allowing e2e upgrade workflow to be triggered manually via UI
- Removing hardcoding of current version and upgrade tags in upgrade tests. 

This can be improved further in follow up PRs once we determine the best path forward for upgrade tests and how to maintain them. For now this provides a means to configure the upgrade test in the workflow yaml.

Target branch: `damian/2188-v6-upgrade-e2e`

part of: #2188 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/10-structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: .github/workflows/e2e-upgrade.yaml [hunk 1]
```diff
@@ -12,7 +9,16 @@ jobs:
       matrix:
         include:
           - test: TestV4ToV5ChainUpgrade
+            chain-image: ghcr.io/cosmos/ibc-go-simd
+            chain-a-tag: v4.0.0
+            chain-b-tag: v4.0.0
+            chain-upgrade-tag: pr-2144 # TODO: backport default upgrade handler to release/v5.x.x
```
Comment (colin-axner) [inline .github/workflows/e2e-upgrade.yaml:15]: is there a pr for this?

File: .github/workflows/e2e-upgrade.yaml [hunk 2]
```diff
@@ -12,7 +9,16 @@ jobs:
...
           - test: TestV5ToV6ChainUpgrade
+            chain-image: ghcr.io/cosmos/ibc-go-icad
+            chain-binary: icad
+            chain-a-tag: v0.3.5
+            chain-b-tag: v0.3.5
+            chain-upgrade-tag: v0.4.0
```
Comment (colin-axner) [inline .github/workflows/e2e-upgrade.yaml:21]: should this be v6.0.0?

**Review by colin-axner — APPROVED** [review]
utACK

File: .github/workflows/e2e-upgrade.yaml (hunk 1 above)
Comment (damiannolan) [reply .github/workflows/e2e-upgrade.yaml:15]: backporting the default upgrade handler here: #2551 #2552

File: .github/workflows/e2e-upgrade.yaml (hunk 2 above)
Comment (damiannolan) [reply .github/workflows/e2e-upgrade.yaml:21]: this is referring to the `icad` docker image version, which includes `ibc-go/v6-alpha1`

**Review by chatton — APPROVED** [review]
LGTM!

**Pull request merged**